hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/ProtobufLogWriter.java

```diff
@@ -85,6 +90,12 @@ public void sync(boolean forceSync) throws IOException {
     } else {
       fsdos.hflush();
     }
+    AtomicUtils.updateMax(this.syncedLength, fsdos.getPos());
```

Comment:
I believe I mentioned elsewhere that I think `AtomicUtils.updateMax` can be replaced with [`AtomicLong.getAndAccumulate`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLong.html#getAndAccumulate-long-java.util.function.LongBinaryOperator-), something like
```
syncedLength.getAndAccumulate(fsdos.getPos(), Math::max)
```

Reply:
Good. Anyway I think this could be another issue to just remove the AtomicUtils class? The current PR is backport so we'd better keep it in sync with the one applied to master.

Reply:
@Apache9 I read your later comments about keeping `AtomicUtils` because the built-in `AtomicLong` does not provide the short-circuit semantics you have implemented there. Sorry, I cannot find that thread to reply there. I dug around and don't see any intrinsics in the JVM that would implement this for us, so I agree with keeping our implementation.